pkg/http/api.go

```diff
@@ -769,9 +769,17 @@ func (a *api) onDirectMessage(reqCtx *fasthttp.RequestCtx) {
 	statusCode := int(resp.Status().Code)
 	if !resp.IsHTTPResponse() {
 		statusCode = invokev1.HTTPStatusFromCode(codes.Code(statusCode))
+		if statusCode != fasthttp.StatusOK {
+			body, err = invokev1.ProtobufToJSON(resp.Status())
+		}
 	}
 
-	respond(reqCtx, statusCode, body)
+	if err != nil {
+		msg := NewErrorResponse("ERR_MALFORMED_REQUEST", err.Error())
+		respondWithError(reqCtx, fasthttp.StatusBadRequest, msg)
```

Comment:
The way this is structured is a little confusing because the only case that can cause `err != nil` is line 773.

Reply:
Is this really the right status? The "malformed request" in this case would be a failure to serialize `resp.Status()` as JSON.

Reply:
You're right. it is to handle L773. I will restructure it. Status code must be Internal Error. I will change it.

Reply:
I was copying the existing code :) I will commit the changes.

Reply:
@ryanvolum Please review it again.

Reply:
You mean @rynowak